cea/demand/ventilation_air_flows_simple.py

```diff
@@ -0,0 +1,196 @@
+        m_ve_mech = max(tsd['m_ve_required'])  # TODO: some night flushing rule
+
+    elif not control_ventilation_systems.is_mechanical_ventilation_active(bpr, tsd, t):
+
+        # mechanical ventilation is turned off
+        m_ve_mech = 0
+
+    else:
+        raise ValueError
+
+    tsd['m_ve_mech'][t] = m_ve_mech
+
+    return
+
+
+def calc_air_mass_flow_window_ventilation(bpr, tsd, t):
+    """
+    Calculates mass flow rate of window ventilation at time step t according to ventilation control options and
+     building systems properties
+
+    Author: Gabriel Happle
+    Date: 01/2017
+
+    :param bpr: Building properties row object
+    :param tsd: Time series data dict
+    :param t: time step [0..8760]
+    :return: updates tsd
+    """
+
+    # if has window ventilation and not special control : m_ve_window = m_ve_schedule
+    if control_ventilation_systems.is_window_ventilation_active(bpr, tsd, t) \
+            and not control_ventilation_systems.is_night_flushing_active(bpr, tsd, t):
+
+        # window ventilation fulfills requirement (control by occupants similar to CO2 sensor)
+        m_ve_window = max(tsd['m_ve_required'][t] - tsd['m_ve_inf_simple'][t], 0)
+        # TODO: check window ventilation calculation, there are some methods in SIA2044
+
+    elif control_ventilation_systems.is_window_ventilation_active(bpr, tsd, t) \
+            and control_ventilation_systems.is_night_flushing_active(bpr, tsd, t):
+
+        # ventilation with maximum capacity = maximum required ventilation rate
+        m_ve_window = max(tsd['m_ve_required'])  # TODO: implement some night flushing rule
+
+    elif not control_ventilation_systems.is_window_ventilation_active(bpr, tsd, t):
+
+        m_ve_window = 0
+
+    else:
+        raise ValueError
+
+    tsd['m_ve_window'][t] = m_ve_window
+
+    return
+
+
+def calc_m_ve_leakage():
+    # requires iteration
+    # TODO: code
+    # TODO: documentation
+    return
+
+
+def calc_m_ve_leakage_simple(bpr, tsd, gv):
+    """
+    Calculates mass flow rate of leakage at time step t according to ventilation control options and
+     building systems properties
+
+    Estimation of infiltration air volume flow rate according to Eq. (3) in DIN 1946-6
+
+    Author: Gabriel Happle
+    Date: 01/2017
+
+    :param bpr: Building properties row object
+    :param tsd: Time series data dict
+    :param gv: globalvars
+    :return: updates tsd
+    """
+
+    # 'flat rate' infiltration considered for all buildings
+
+    # get properties
+    n50 = bpr.architecture['n50']
+    area_f = bpr.rc_model['Af']
+
+    # estimation of infiltration air volume flow rate according to Eq. (3) in DIN 1946-6
+    n_inf = 0.5 * n50 * (gv.delta_p_dim/50) ** (2/3)  # [air changes per hour] m3/h.m2
+    infiltration = gv.hf * area_f * n_inf * 0.000277778  # m3/s
+
+    tsd['m_ve_inf_simple'] = infiltration * physics.calc_rho_air(tsd['T_ext'][:])  # (kg/s)
```

Comment:
@daren-thomas @JIMENOFONSECA the question is whether this is faster. I can try. This might also be one of the reasons why the code is slower. I execute some functions many times per time step instead of looking up previously calculated results. What do you think Daren?